tools/swift-format/Sources/Rules/NoAccessLevelOnExtensionDeclaration.swift

```diff
@@ -10,7 +10,148 @@ import SwiftSyntax
+          let newDecl = varDecl.withModifiers(modifiers.replacing(childAt: next.indexInParent, with: formattedNext).prepending(formattedKeyword))
+          newMembers.append(member.withDecl(newDecl))
+        }
+      
+      } else if let funcDecl = member.decl as? FunctionDeclSyntax {
+        guard let modifiers = funcDecl.modifiers
+          else {
+            let formattedFuncDecl = funcDecl.withFuncKeyword(funcDecl.funcKeyword.withoutLeadingTrivia())
+            newMembers.append(member.withDecl(formattedFuncDecl.addModifier(formattedKeyword)))
+            continue
+        }
+        guard hasAccessorKeyword(modifiers: modifiers) == false
+          else { newMembers.append(member); continue }
+        
+        if let next = modifiers.first {
+          let formattedNext = next.withName(next.name.withoutLeadingTrivia())
+          let newDecl = funcDecl.withModifiers(modifiers.replacing(childAt: next.indexInParent, with: formattedNext).prepending(formattedKeyword))
+          newMembers.append(member.withDecl(newDecl))
+        }
+      
+        
+        // TODO(laurenwhite)
+      } else if let associatedTypeDecl = member.decl as? AssociatedtypeDeclSyntax {
+        let formattedAssociateDecl = associatedTypeDecl.withAssociatedtypeKeyword(associatedTypeDecl.associatedtypeKeyword.withoutLeadingTrivia())
+        newMembers.append(member.withDecl(formattedAssociateDecl.addModifier(formattedKeyword)))
+      
+      } else if let classDecl = member.decl as? ClassDeclSyntax {
+        let formattedClassDecl = classDecl.withClassKeyword(classDecl.classKeyword.withoutLeadingTrivia())
+        newMembers.append(member.withDecl(formattedClassDecl.addModifier(formattedKeyword)))
+      
+      } else if let enumDecl = member.decl as? EnumDeclSyntax {
+        let formattedEnumDecl = enumDecl.withEnumKeyword(enumDecl.enumKeyword.withoutLeadingTrivia())
+        newMembers.append(member.withDecl(formattedEnumDecl.addModifier(formattedKeyword)))
+      
+      } else if let protocolDecl = member.decl as? ProtocolDeclSyntax {
+        let formattedProtocolDecl = protocolDecl.withProtocolKeyword(protocolDecl.protocolKeyword.withoutLeadingTrivia())
+        newMembers.append(member.withDecl(formattedProtocolDecl.addModifier(formattedKeyword)))
+      
+      } else if let structDecl = member.decl as? StructDeclSyntax {
+        let formattedStructDecl = structDecl.withStructKeyword(structDecl.structKeyword.withoutLeadingTrivia())
+        newMembers.append(member.withDecl(formattedStructDecl.addModifier(formattedKeyword)))
+      
+      } else if let typeAliasDecl = member.decl as? TypealiasDeclSyntax {
+        let formattedAliasDecl = typeAliasDecl.withTypealiasKeyword(typeAliasDecl.typealiasKeyword.withoutLeadingTrivia())
+        newMembers.append(member.withDecl(formattedAliasDecl.addModifier(formattedKeyword)))
+      
+      } else if let initDecl = member.decl as? InitializerDeclSyntax {
```

Comment:
It doesn't seem like SwiftSyntax has a better way to do this...you might actually want to build another Rewriter whose job is to prepend a modifier onto a decl, and then override the methods for each of these types inside that rewriter.

```swift
class AddModifierRewriter: SyntaxRewriter {
  let modifierKeyword: TokenSyntax

  override visit(_ function: FunctionDeclSyntax) -> DeclSyntax { ... }
}
```